core/src/apps/cardano/sign_tx.py

```diff
@@ -787,23 +954,39 @@ async def _show_pool_owner(
             POOL_OWNER_STAKING_PATH_NAME,
         )
 
-    await confirm_stake_pool_owner(ctx, keychain, owner, network_id)
+    await confirm_stake_pool_owner(ctx, keychain, owner, protocol_magic, network_id)
 
 
 def _validate_witness_request(
     witness_request: CardanoTxWitnessRequest,
     signing_mode: CardanoTxSigningMode,
+    transaction_has_token_minting: bool,
     account_path_checker: AccountPathChecker,
 ) -> None:
-    # witness path validation happens in _show_witness
+    # further witness path validation happens in _show_witness_request
+    is_minting = SCHEMA_MINT.match(witness_request.path)
 
-    if signing_mode == CardanoTxSigningMode.POOL_REGISTRATION_AS_OWNER:
-        _ensure_no_payment_witness(witness_request)
+    if signing_mode == CardanoTxSigningMode.ORDINARY_TRANSACTION:
+        if not (
+            is_byron_path(witness_request.path)
+            or is_shelley_path(witness_request.path)
+            or is_minting
+        ):
+            raise INVALID_WITNESS_REQUEST
+        if is_minting and not transaction_has_token_minting:
+            raise INVALID_WITNESS_REQUEST
+    elif signing_mode == CardanoTxSigningMode.POOL_REGISTRATION_AS_OWNER:
+        _ensure_only_staking_witnesses(witness_request)
+    elif signing_mode == CardanoTxSigningMode.MULTISIG_TRANSACTION:
+        if not is_multisig_path(witness_request.path) and not is_minting:
```

Comment:
With the rename, minting falls under "multisig" transaction.

Is this correct or an oversight?

Reply:
Minting witnesses can be used both under `MULTISIG` and `ORDINARY`. The only condition is that minting is actually present in the transaction. Minting witness paths are always shown to the user.